src/util/psbt/serialize.rs

```diff
@@ -141,3 +145,28 @@ impl Deserialize for SigHashType {
         }
     }
 }
+
+// global xpubs
+impl Deserialize for ExtendedPubKey {
+    fn deserialize(bytes: &[u8]) -> Result<Self, encode::Error> {
+        let xpub_str = base58::check_encode_slice(bytes);
+        let xpub = Self::from_str(&xpub_str).unwrap();
+        Ok(xpub)
+    }
+}
+
+impl Serialize for ExtendedPubKey {
+    fn serialize(&self) -> Vec<u8> {
+        let mut ret = [0; 78];
+        ret[0..4].copy_from_slice(&match self.network {
+            Network::Bitcoin => [0x04u8, 0x88, 0xB2, 0x1E],
+            Network::Testnet | Network::Signet | Network::Regtest => [0x04u8, 0x35, 0x87, 0xCF],
+        }[..]);
+        ret[4] = self.depth as u8;
+        ret[5..9].copy_from_slice(&self.parent_fingerprint[..]);
+        ret[9..13].copy_from_slice(&endian::u32_to_array_be(u32::from(self.child_number)));
+        ret[13..45].copy_from_slice(&self.chain_code[..]);
+        ret[45..78].copy_from_slice(&self.public_key.key.serialize()[..]);
+        ret.to_vec()
+    }
+}
```

Comment:
This code is mostly copied from the `Display` impl for `ExtendedPubKey`. I'm sure there's a better way ...